components/molecule/buttonGroup/src/index.js

```diff
@@ -46,15 +47,20 @@ const MoleculeButtonGroup = ({
         isFunction(onClickChild) && onClickChild(event, ...args)
         isFunction(onClick) && onClick(event, ...args)
       }
-      return cloneElement(child, {
-        ...props,
-        negative: combineProps(negativeChild, negativeProp),
-        size: combineProps(sizeChild, sizeProp),
-        design: combineProps(designChild, designProp),
-        groupPosition,
-        fullWidth,
-        onClick: clickHandler
-      })
+
+      return (
+        <Injector
+          {...props}
+          negative={combineProps(negativeChild, negativeProp)}
+          size={combineProps(sizeChild, sizeProp)}
+          design={combineProps(designChild, designProp)}
+          groupPosition={groupPosition}
+          fullWidth={fullWidth}
+          onClick={clickHandler}
+        >
+          {child}
+        </Injector>
```

Comment:
The `Injector` component manages the props combination between children and injected, so it might be delegated to them. As a result yo can only pass:
```js
  <Injector
    {...props}
    negative={negativeProp} // or simply negative with no renaming
    size={sizeProp} // or simply size with no renaming
    design={designProp} // or simply design with no renaming
    groupPosition={groupPosition}
    fullWidth={fullWidth}
    onClick={clickHandler} // directly the onClick, cuz it manages the handlers also, and clickHandler can be removed
  >
    {child}
  </Injector>
```

Reply:
propbably the combineProp declared in settings can also be removed